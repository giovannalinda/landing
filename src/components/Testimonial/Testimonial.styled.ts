import styled from 'styled-components'
import { theme } from '~/styles/theme'

type IndicatorProps = {
  top: number
}

export const Container = styled.section`
  padding-top: 18rem;
  padding-bottom: 10rem;
`

export const TestimonialGroup = styled.div`
  display: grid;
  grid-template-columns: 1fr 3fr;
  grid-gap: 4.4rem;
  margin-top: ${theme.space[8]};
  color: ${theme.colors.primaryDark};

  @media screen and (max-width: ${theme.breakpoints.mobile}) {
    grid-template-columns: 1fr;
    grid-gap: ${theme.space[5]};
  }
`

export const TestimonialList = styled.ul`
  display: grid;
  grid-template-rows: repeat(4, 1fr);
  grid-gap: ${theme.space[0]};

  @media screen and (max-width: ${theme.breakpoints.mobile}) {
    display: flex;
    overflow-x: scroll;
    align-items: center;
    grid-gap: ${theme.space[3]};
  }
`

export const QuoteContainer = styled.div`
  padding: ${theme.space[9]};
  position: relative;
  background: ${theme.colors.white};
  border-radius: ${theme.borderRadius.md};

  span {
    font-size: ${theme.font.sizes.large};
    padding-left: 1rem;
    display: flex;
    align-items: flex-end;
    color: ${theme.colors.primaryDark};

    svg {
      color: ${theme.colors.secondaryOrange};
      margin-right: ${theme.space[1]};
    }
  }

  p {
    padding-top: 2.6rem;
    font-size: ${theme.font.sizes.subtitle};
    line-height: 5.1rem;
    color: ${theme.colors.primaryDark};
  }

  @media screen and (max-width: ${theme.breakpoints.mobile}) {
    p {
      font-size: ${theme.font.sizes.large};
      line-height: ${theme.font.lineHeights.large};
    }
  }
`

export const QuoteImageWrapper = styled.div`
  position: absolute;
  right: ${theme.space[5]};
  top: ${theme.space[5]};
`

export const Indicator = styled.div<IndicatorProps>`
  position: absolute;
  height: 3.2rem;
  width: 3.2rem;
  background: ${theme.colors.white};
  left: -1.6rem;
  top: ${({ top }) => `${top}px`};
  transform: rotate(-45deg);
  transition: all ${theme.transitions.slow};
  border-radius: ${theme.borderRadius.sm} 0 0 0;

  @media screen and (max-width: ${theme.breakpoints.mobile}) {
    display: none;
  }
`
