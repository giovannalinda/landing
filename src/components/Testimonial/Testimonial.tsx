import { useRef } from 'react'
import { BsFillStarFill } from 'react-icons/bs'
import Image from 'next/image'

import { Assets } from '~/assets'
import { Title } from '~/components'
import { useIndicator } from '~/lib/useIndicator'

import { TestimonialCard } from './TestimonialCard'

import * as S from './Testimonial.styled'

const people = [
  {
    name: 'Sundar Pichai',
    office: 'CEO e Founder',
    company: 'Google',
  },
  {
    name: 'Shantanu Narayen',
    office: 'CEO',
    company: 'Adobe',
  },
  {
    name: 'Noah Levin',
    office: 'Director of Design',
    company: 'Figma',
  },
  {
    name: 'Stewart Butterfield',
    office: 'CEO e Founder',
    company: 'Slack',
  },
]

const VALUE_TO_CENTER = 35

export function Testimonial() {
  const testimonialListRef = useRef(null)

  const { indicatorPosition, changeIndicatorPosition } = useIndicator(
    testimonialListRef,
    VALUE_TO_CENTER,
  )

  return (
    <S.Container>
      <Title>Testimonial</Title>

      <S.TestimonialGroup>
        <S.TestimonialList ref={testimonialListRef}>
          {people.map((person) => (
            <li key={person.name} onClick={changeIndicatorPosition}>
              <TestimonialCard
                name={person.name}
                office={person.office}
                company={person.company}
              />
            </li>
          ))}
        </S.TestimonialList>

        <S.QuoteContainer>
          <span>
            <BsFillStarFill size={24} />
            5.0 Star Rating
          </span>
          <S.QuoteImageWrapper>
            <Image src={Assets.Quote} width='95' height='75' />
          </S.QuoteImageWrapper>
          <p>
            “If you&apos;re looking for someone who will challenge your UX/UI
            thinking and really cut to the core of what&apos;s important for
            users, then Jesse is your man. On top of that, he brings a level of
            enthusiasm to the craft that&apos;s energizing for everyone who
            works with him.”
          </p>

          <S.Indicator top={indicatorPosition} />
        </S.QuoteContainer>
      </S.TestimonialGroup>
    </S.Container>
  )
}
