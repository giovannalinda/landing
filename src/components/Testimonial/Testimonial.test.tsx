import { render } from '@testing-library/react'

import { Testimonial } from './Testimonial'

describe('Testimonial', () => {
  it('should match snapshot', () => {
    const { container } = render(<Testimonial />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
