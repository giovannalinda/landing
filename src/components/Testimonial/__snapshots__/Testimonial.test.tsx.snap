// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testimonial should match snapshot 1`] = `
<section
  class="sc-bBXxYQ cbPxtP"
>
  <h1
    class="sc-hAZoDl cKtAR"
  >
    Testimonial
  </h1>
  <hr
    class="sc-fnykZs hxMeGu"
  />
  <div
    class="sc-cOFTSb fTWQsl"
  >
    <ul
      class="sc-BeQoi fUsbec"
    >
      <li>
        <button
          class="sc-cTQhss kagvef"
        >
          <div
            class="sc-jOrMOR evvYZV"
          />
          <div
            class="sc-dPyBCJ IWFCe"
          >
            <h3>
              Sundar Pichai
            </h3>
            <small
              aria-label="CEO e Founder of Google"
            >
              CEO e Founder
               of 
              <span>
                Google
              </span>
            </small>
          </div>
        </button>
      </li>
      <li>
        <button
          class="sc-cTQhss kagvef"
        >
          <div
            class="sc-jOrMOR evvYZV"
          />
          <div
            class="sc-dPyBCJ IWFCe"
          >
            <h3>
              Shantanu Narayen
            </h3>
            <small
              aria-label="CEO of Adobe"
            >
              CEO
               of 
              <span>
                Adobe
              </span>
            </small>
          </div>
        </button>
      </li>
      <li>
        <button
          class="sc-cTQhss kagvef"
        >
          <div
            class="sc-jOrMOR evvYZV"
          />
          <div
            class="sc-dPyBCJ IWFCe"
          >
            <h3>
              Noah Levin
            </h3>
            <small
              aria-label="Director of Design of Figma"
            >
              Director of Design
               of 
              <span>
                Figma
              </span>
            </small>
          </div>
        </button>
      </li>
      <li>
        <button
          class="sc-cTQhss kagvef"
        >
          <div
            class="sc-jOrMOR evvYZV"
          />
          <div
            class="sc-dPyBCJ IWFCe"
          >
            <h3>
              Stewart Butterfield
            </h3>
            <small
              aria-label="CEO e Founder of Slack"
            >
              CEO e Founder
               of 
              <span>
                Slack
              </span>
            </small>
          </div>
        </button>
      </li>
    </ul>
    <div
      class="sc-kgUAyh eCKseL"
    >
      <span>
        <svg
          fill="currentColor"
          height="24"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.282.95l-3.522 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
          />
        </svg>
        5.0 Star Rating
      </span>
      <div
        class="sc-hTtwUo hGyANs"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2795%27%20height=%2775%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <p>
        “If you're looking for someone who will challenge your UX/UI thinking and really cut to the core of what's important for users, then Jesse is your man. On top of that, he brings a level of enthusiasm to the craft that's energizing for everyone who works with him.”
      </p>
      <div
        class="sc-jgbSNz kHahAz"
      />
    </div>
  </div>
</section>
`;
